### PR TITLE
Port changes

### PR DESCRIPTION
**Pull request by SchnWalter**

### CODE REVIEW

File: src/common/email/email.module.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      provide: EMAIL_SERVICE,
+      useClass: EmailService,
+    },
+  ],
+  controllers: [],
+  exports: [
+    EMAIL_SERVICE,
+  ],
+})
+export class EmailModule {}
```
Comment (mcristi) [inline src/common/email/email.module.ts]: let's pluralize this module and its services

File: src/common/users/services/users.interface.ts [hunk 2]
```diff
@@ -1,10 +1,13 @@
+import { DeepPartial } from 'typeorm';
+
 import { CreateUserDto } from '../dto/create-user.dto';
 import { User } from '../entities/user.entity';
 
 export interface IUsersService {
-  create(createUserDto: CreateUserDto);
+  create(createUserDto: CreateUserDto): Promise<User | null>;
```
Comment (mcristi) [inline src/common/users/services/users.interface.ts]: This method should return only `User` objects, never null.

File: src/common/users/services/users.service.ts [hunk 3]
```diff
@@ -1,42 +1,34 @@
...
-    const user: User = await this.userRepository.save(createUser);
-
-    return {
-      message: 'User created successfully.',
-      user: user,
-    };
+    return this.userRepository.save(createUser);
   }
 
   findById(id: string): Promise<User | null> {
```
Comment (mcristi) [inline src/common/users/services/users.service.ts]: we should use `findOneOrFail` here

File: src/common/users/services/users.service.ts [hunk 4]
```diff
@@ -53,7 +45,16 @@ export class UsersService implements IUsersService {
...
+      email: email.toLowerCase(),
     });
   }
+
+  async update(userId: string, user: DeepPartial<User>): Promise<User> {
+    if (user.email) {
+      user.email = user.email.toLowerCase();
+    }
+    await this.userRepository.update(userId, user);
+    return await this.findById(userId) as User;
```
Comment (mcristi) [inline src/common/users/services/users.service.ts]: if in `findById` is used `findOneOrFail`, `as User` could be removed

File: src/config/security.config.ts [hunk 5]
```diff
@@ -1,12 +1,27 @@
...
+export interface SecurityConfiguration {
+  accessTokenSecret: string;
+  accessTokenExpirationTime: string;
+  refreshTokenSecret: string;
+  refreshTokenExpirationTime: string;
+  genericTokenSecret: string;
+  genericTokenExpirationTimeInHours: string;
+  tokensIssuer: string;
+  tokensAlgorithm: string;
+}
```
Comment (mcristi) [inline src/config/security.config.ts:14]: If we have an interface now, please update all the usages of the security configurations to use the interface.

File: src/common/users/services/users.service.ts [hunk 6]
```diff
@@ -1,42 +1,34 @@
...
     private readonly userRepository: Repository<User>,
-  ) {}
+  ) {
+  }
 
   async create(createUserDto: CreateUserDto) {
     if (await this.findByUsername(createUserDto.username)) {
-      this.logger.warn(USERNAME_ALREADY_EXISTS_ERROR_MESSAGE);
-      throw new BadRequestException(USERNAME_ALREADY_EXISTS_ERROR_MESSAGE);
+      this.logger.warn('SERVER_ERROR.USER.USERNAME_ALREADY_EXISTS');
```
Comment (mcristi) [inline src/common/users/services/users.service.ts]: why not a file const if it used in multiple places? a const will ease the refactoring

File: src/modules/accounts/services/accounts.service.ts [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+      throw new UnauthorizedException();
+    }
+
+    const user: User | null = await this.usersService.findById(userId);
+    if (!user) {
+      throw new UnauthorizedException();
+    }
+
+    if (dto.newPassword !== dto.confirmPassword) {
+      throw new BadRequestException('SERVER_ERROR.USER.PASSWORDS_DO_NOT_MATCH');
```
Comment (mcristi) [inline src/modules/accounts/services/accounts.service.ts]: same, I would prefere to have the keys kept in file constants

File: src/modules/accounts/services/accounts.service.ts [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+    try {
+      const payload: JwtPayload = await this.authService.verifyGenericToken(dto.token);
+      userId = payload.sub;
+    } catch (err) {
+      throw new UnauthorizedException();
+    }
+
+    const user: User | null = await this.usersService.findById(userId);
+    if (!user) {
+      throw new UnauthorizedException();
```
Comment (mcristi) [inline src/modules/accounts/services/accounts.service.ts]: This check is not needed if `findById` will fail anyway at `findOneOrFail`

File: src/modules/accounts/services/accounts.service.ts [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+    @Inject(AUTH_SERVICE)
+    private readonly authService: IAuthService,
+    @Inject(USERS_SERVICE)
+    private readonly usersService: IUsersService,
+  ) {
+  }
+
+  public async forgotPassword(dto: ForgotPasswordDto): Promise<void> {
+    const user = await this.usersService.findByEmail(dto.email);
+    if (user && user.active) {
```
Comment (mcristi) [inline src/modules/accounts/services/accounts.service.ts]: let's rename `active` to `isActive` ... it's easier to read in a natural way.

File: src/modules/users/index.ts [hunk 10]
```diff
@@ -1,5 +1 @@
 export * from './users.module';
```
Comment (mcristi) [inline src/modules/users/index.ts:1]: being a common module, it will be used by many others, so let's keep `'./services/users.interface` exported

**Review by mcristi — CHANGES_REQUESTED** [review]

File: src/modules/users/index.ts (hunk 10 above)
Comment (SchnWalter) [reply src/modules/users/index.ts:1]: This module only contains the API endpoints controller, I've moved that code into `~common/users`.

File: .eslintrc.js [hunk 11]
```diff
@@ -20,9 +20,13 @@ module.exports = {
       {
         "ignore": [
           "^~[^/]+/[^/]+$",
+          "@sendgrid/helpers",
+          "@sendgrid/mail",
+          "jsonwebtoken",
         ]
       }
     ],
+    "jsdoc/newline-after-description": "off",
```
Comment (mcristi) [inline .eslintrc.js:29]: Isn't this a rule in the lint?

File: src/common/users/services/users.interface.ts [hunk 12]
```diff
@@ -1,10 +1,14 @@
+import { DeepPartial } from 'typeorm';
+
 import { CreateUserDto } from '../dto/create-user.dto';
 import { User } from '../entities/user.entity';
 
 export interface IUsersService {
-  create(createUserDto: CreateUserDto);
+  create(createUserDto: CreateUserDto): Promise<User>;
   findById(id: string): Promise<User | null>;
+  findByIdOrFail(id: string): Promise<User>;
```
Comment (mcristi) [inline src/common/users/services/users.interface.ts]: `findById` should have `findByIdOrFail` behaviour by default. We should not have `orFail` methods exposed by services. If we expect to fail, we should catch the error, but it's not the case anywhere. 
Let's keep only `findById` with fail behaviour

File: src/modules/accounts/services/accounts.service.ts (hunk 7 above)
Comment (mcristi) [reply src/modules/accounts/services/accounts.service.ts]: still not resolved

**Review by mcristi — CHANGES_REQUESTED** [review]

**Review by mcristi — APPROVED** [review]

**Pull request merged**